Headless testing with poltergeist

basiszwo opened the pull request.

basiszwo committed:
Add poltergeist as default driver for capybara

basiszwo committed:
Use selenium and do not maximize window in work package spec

basiszwo committed:
Do not maximize window

basiszwo committed:
Use selenium driver for some specs

basiszwo committed:
Remove maximize window context

basiszwo committed:
Switch cukes to poltergeist (selectively run selenium)

Some features rely on selenium only features which need some more time
to work. Therefore we switch back to selenium selectively

basiszwo committed:
Switch wiki feature test to selenium

basiszwo committed:
Stabilize timelines cucumber feature

basiszwo committed:
Wip out cukes with undefined / pending steps

basiszwo committed:
Remove redundant move work package cuke

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://basiszwo-headless-testing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://basiszwo-headless-testing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

basiszwo committed:
Resize window size for poltergeist to 1200x1000

basiszwo committed:
Use selenium as interaction with alerts does not work with poltergeist

basiszwo committed:
Use rspec built in functionality to rerun failed specs

This is basically the same as with cucumber. Rerunning specs in a new
process is less error prone as you'll get a complete new environment.

basiszwo committed:
Add script to run and rerun testsuite on CI servers

We run the suites (cucumber, rspec) and log failures. In a second run
we just execute the failing ones.

basiszwo committed:
Increase poltergeist timeout

basiszwo committed:
Support local parallel testing for ci_runner.sh script

basiszwo committed:
Do not require legacy rspec formatters by default

basiszwo committed:
Update travis configuration

* parallelize build matrix
* use phantomjs 2.0
* use postgres 9.3
* run custom scripts
* whiteliste main branches

basiszwo committed:
Use parallel test and rubocop gem in development and test env

basiszwo committed:
Fix mysql database name, add proper postgres database yml

basiszwo committed:
Run specs and cukes based on runtime logs

oliverguenther commented:
Just saw this passing on travis. Awesome! :+1:

basiszwo commented:
still lots of work to do. Currently we are running some in selenium mode but we're getting there :wink:

basiszwo committed:
Use selenium until we upgraded jenkins CI workers to phantomjs 2.0

basiszwo committed:
Remove unused ci rake tasks

basiszwo commented:
@why-el can you please review and merge? I needed to disable poltergeist (again) as we need to update jenkins first. this can be done next week.

why-el commented:
Ok, on it.

why-el commented on script/files/parallel_runtime_rspec.log:
```diff
@@ -0,0 +1,443 @@
+spec/controllers/work_packages/bulk_controller_spec.rb:99.308038
```
Is this .log file necessary? Same with the one above.

basiszwo replied:
I will improve this at a later point to re-write it with every run and download it from somewhere. I have it on my list.
